projecttacoma/fqm-testify
# Mrs input

# Summary
FQM-Testify can now import a Measure from a passed in Measure Repository Service URL

## New Behavior
All old behavior should be unchanged!
There is now a new option on the Measure Upload page to upload from Measure Repository Service
The app should work identically once a valid Measure is uploaded through this new method.

## Code Changes

- Created MeasureRepositoryUpload.tsx, a component that pulls measures from a MRS and then packages the selected measure with all dependencies into a Measure Bundle to pass into testify
- Created MeasureRepositoryHeader.tsx, a component that frames the above component and adds helpful information
- Created displayMapToSelectData Selector to populate Measure Select component.
- Added additional information to MeasureBundleState to determine the type of upload and persist information for both file and MRS sourced upload when leaving/returning to the page
- Minor updates to other components that access MeasureBundleState to accommodate these changes
- Minor updates to existing tests to accommodate changes

# Testing Guidance
- `npm run check`
- Run through the regular file upload workflow and ensure nothing has changed
- Start up our Measure Repository Service and upload as many valid and invalid bundles as you'd like
- Start up Testify with `npm run dev`
- Type in the url of the running measure repository service and ensure the call is successful and the measure select dropdown appears and is populated accurately
- Select a working measure and ensure that Testify workflow is unchanged from there
- Select a broken measure and ensure that the error handling is intuitive and properly labeled
- Try to break the system. Move back and forth between screens. Upload a file when a bundle is already uploaded through MRS and vice versa. Change the url after uploading a bundle. Make sure the code works intuitively throughout the process.
